source/jormungandr/tests/line_sections_tests.py

```diff
@@ -165,90 +249,211 @@ def test_traffic_reports_on_stop_areas(self):
                 ObjGetter('stop_areas', sa),
             )
 
+        # line_section_on_line_2
+        scenario = {'A': False, 'B': True, 'C': False, 'D': False, 'E': False, 'F': True}
+
+        for sa, result in scenario.iteritems():
+            assert result == self.has_tf_disruption(
+                'stop_areas/{}/traffic_reports'.format(sa), 'line_section_on_line_2'
+            )
+            assert result == self.has_tf_linked_disruption(
+                'stop_areas/{}/traffic_reports'.format(sa), 'line_section_on_line_2', ObjGetter('stop_areas', sa)
+            )
+
     def test_traffic_reports_on_networks(self):
-        def has_dis(q):
-            r = self.default_query(q)
-            return 'line_section_on_line_1' in (d['disruption_id'] for d in r['disruptions'])
+        for disruption_label in [
+            'line_section_on_line_1',
+            'line_section_on_line_1_other_effect',
+            'line_section_on_line_2',
+        ]:
+            assert self.has_dis('networks/base_network/traffic_reports', disruption_label)
+            assert not self.has_dis('networks/network:other/traffic_reports', disruption_label)
 
-        assert has_dis('networks/base_network/traffic_reports')
-        assert not has_dis('networks/network:other/traffic_reports')
         r = self.default_query('traffic_reports')
-        assert len(r['traffic_reports']) == 1  # only one network (base_network) is disrupted
-        assert len(r['traffic_reports'][0]['stop_areas']) == 4
+        # only one network (base_network) is disrupted
+        assert len(r['traffic_reports']) == 1
+        assert len(r['traffic_reports'][0]['stop_areas']) == 5
         for sa in r['traffic_reports'][0]['stop_areas']:
-            assert len(sa['links']) == (2 if sa['id'] == 'E' else 1)
+            assert len(sa['links']) == (2 if sa['id'] in ['A', 'E', 'F'] else 1)
 
+    @pytest.mark.xfail(
+        strict=True,
+        reason="the disruption affects at least 1 stop of the line whereas the line itself is not impacted",
+        raises=AssertionError,
+    )
     def test_traffic_reports_on_lines(self):
         """
         we should be able to find the related line section disruption with /traffic_report
         """
 
         # line_section_on_line_1
-        assert self.has_tf_disruption('lines/line:1/traffic_reports', 'line_section_on_line_1')
-        assert self.has_tf_linked_disruption(
-            'lines/line:1/traffic_reports', 'line_section_on_line_1', ObjGetter('lines', 'line:1')
-        )
+        scenario = {
+            'line:1': True,
+            'line:2': False,
+            'line:3': False  # fail
```

Comment:
What I fear now is we won't realize line:1 or line:2 test failed as it's an xfail test as a whole.
But then it becomes more painfull to use xfail if that prevent testcode-mutualization...